cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -425,6 +425,15 @@ public Void run(SentryGenericServiceClient client) throws Exception {
         if (principal.getType().equals(Principal.PrincipalType.USER)) {
           // for a user get all the groups and their roles
           Set<String> groups = authProvider.getGroupMapping().getGroups(principal.getName());
+          LOG.debug("Got groups {} for principal {}", groups, principal);
+          if (!groups.contains(principal.getName())) {
+            LOG.debug("Principal {} does not belong to group {}. Adding the principal's group {} to the groups for " +
+                        "enforcement",
```

Comment:
If cannot avoid line break for the debug message, wrap it in `isDebugEnabled()` condition.